nexus/src/app/sagas/snapshot_create.rs

```diff
@@ -137,6 +137,10 @@ pub(crate) struct Params {
 // snapshot create saga: actions
 declare_saga_actions! {
     snapshot_create;
+    TAKE_VOLUME_LOCK -> "volume_lock" {
```

Comment:
We don't also need a volume lock step on the region_replacement saga?
Is that becuase the `volume_repair_delete_query()` does the locking for us?

Reply:
When a region replacement request is created, the associated volume repair record is created at the same time. It's deleted when the region replacement request transitions to Completed.